api/src/data.rs

```diff
@@ -0,0 +1,110 @@
+use crate::{context::Context, metrics::metrics, param::AddressParam};
+use anyhow::Result;
+use warp::{filters::BoxedFilter, Filter, Rejection, Reply};
+
+// GET /data/accounts/<account_id>
+pub fn data_get_account(context: Context) -> BoxedFilter<(impl Reply,)> {
+    warp::path!("data" / "accounts" / AddressParam)
+        .and(warp::get())
+        .and(context.filter())
+        .and_then(get_account)
+        .with(metrics("data_get_account"))
+        .boxed()
+}
+
+// GET /data/accounts/<account_id>/resources
+pub fn data_get_account_resources(context: Context) -> BoxedFilter<(impl Reply,)> {
+    warp::path!("data" / "accounts" / AddressParam / "resources")
+        .and(warp::get())
+        .and(context.filter())
+        .and(warp::query::<Version>())
+        .map(|address, ctx, version: Version| (version.version, address, ctx))
+        .untuple_one()
+        .and_then(get_account_resources)
+        .with(metrics("data_get_account_resources"))
+        .boxed()
+}
+
+// GET /data/accounts/<account_id>/modules
+pub fn data_get_account_modules(context: Context) -> BoxedFilter<(impl Reply,)> {
+    warp::path!("data" / "accounts" / AddressParam / "modules")
+        .and(warp::get())
+        .and(context.filter())
+        .and(warp::query::<Version>())
+        .map(|address, ctx, version: Version| (version.version, address, ctx))
+        .untuple_one()
+        .and_then(get_account_modules)
+        .with(metrics("data_get_account_modules"))
+        .boxed()
+}
+
+// GET /data/accounts/<account_id>/transactions?start={u64}&limit={u16}
+pub fn data_get_account_transactions(context: Context) -> BoxedFilter<(impl Reply,)> {
+    warp::path!("data" / "accounts" / AddressParam / "transactions")
+        .and(warp::get())
+        .and(warp::query::<Page>())
+        .and(context.filter())
+        .and_then(get_account_transactions)
+        .with(metrics("data_get_account_transactions"))
+        .boxed()
+}
+
+// GET /transactions?start={u64}&limit={u16}
+pub fn data_get_transactions(context: Context) -> BoxedFilter<(impl Reply,)> {
+    warp::path!("data" / "transactions")
+        .and(warp::get())
+        .and(warp::query::<Page>())
+        .and(context.filter())
+        .and_then(get_transactions)
+        .with(metrics("data_get_transactions"))
+        .boxed()
+}
+
+async fn get_account(address: AddressParam, context: Context) -> Result<impl Reply, Rejection> {
+    Ok(Account::new(None, address, context)?.raw_account_resource()?)
+}
+
+async fn get_account_resources(
+    ledger_version: Option<LedgerVersionParam>,
+    address: AddressParam,
+    context: Context,
+) -> Result<impl Reply, Rejection> {
+    Ok(Account::new(ledger_version, address, context)?.raw_resources()?)
+}
+
+async fn get_account_modules(
+    ledger_version: Option<LedgerVersionParam>,
+    address: AddressParam,
+    context: Context,
+) -> Result<impl Reply, Rejection> {
+    Ok(Account::new(ledger_version, address, context)?.raw_modules()?)
+}
```

Comment:
maybe just make this inline rather than having another callout here?

Reply:
The callout here is making error type conversion easier.  Trust me this made it a lot easier